### PR TITLE
Replace npm with pnpm

### PR DESCRIPTION
**Pull request by NigelBreslaw**
Replaces the use of NPM with PNPM. This should make it quick and easy to update any of the dependencies as dependant will now see everything in one go. It might make some of the CI items run a little faster.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -130,27 +130,32 @@ jobs:
               run: choco upgrade llvm
             - name: Setup headless display
               uses: pyvista/setup-headless-display-action@v1
-            - uses: ./.github/actions/install-nodejs
```
Comment (hunger) [inline .github/workflows/ci.yaml:133]: I think this action is unused now and can get removed.

That action did some extra steps. Is it save to remove those? E.g. install node-gyp on Windows?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -130,27 +130,32 @@ jobs:
...
+              with:
+                version: 9.11.0
             - uses: ./.github/actions/setup-rust
               with:
                   key: x-napi-v2-${{ steps.node-install.outputs.node-version }} # the cache key consists of a manually bumpable version and the node version, as the cached rustc artifacts contain linking information where to find node.lib, which is in a versioned directory.
-            - name: Run npm install
-              working-directory: ./api/node
-              run: npm install --ignore-scripts
+            - name: Run pnpm install
+              working-directory: api/node
```
Comment (hunger) [inline .github/workflows/ci.yaml:144]: Why did you remove the `./`?

I do not mind you removing that, but then please do it consistently:-)

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -130,27 +130,32 @@ jobs:
...
-              working-directory: ./api/node
+              run: pnpm build:testing
+              working-directory: api/node
             - name: Lint node api
-              working-directory: ./api/node
-              run: npm run lint
+              working-directory: api/node
+              run: pnpm lint
             - name: Check code formatting
               if: ${{ runner.os != 'Windows' }}
```
Comment (hunger) [inline .github/workflows/ci.yaml:153]: Do we really need to run this on Mac and Linux? Maybe this would be better in build-docs.yaml which also does the rust and C++ formatting?

Or maybe -- considering that formatting is not docs related at all -- somewhere else entirely?

That's just a comment that can be fixed later of course.

File: .github/workflows/nightly_snapshot.yaml [hunk 4]
```diff
@@ -236,11 +236,14 @@ jobs:
...
         steps:
             - uses: actions/checkout@v4
+            - uses: pnpm/action-setup@v4.0.0
+              with:
+                version: 9.11.0
             - name: Install GNU Sed
               run: brew install gnu-sed
             - uses: actions/setup-node@v4
               with:
-                  node-version: 18
```
Comment (hunger) [inline .github/workflows/nightly_snapshot.yaml:243]: Maybe we should keep our install-node action, so that we have one central place to set the node-version? We'd need to use that more consistently of course ;-)

Not relevant for this PR.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "gb-slint",
```
Comment (hunger) [inline package.json:2]: What does "gb" stand for?

**Review by hunger — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (NigelBreslaw) [reply .github/workflows/ci.yaml:133]: done

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (NigelBreslaw) [reply .github/workflows/ci.yaml:144]: done

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (NigelBreslaw) [reply .github/workflows/ci.yaml:153]: It takes 5ms to run

File: .github/workflows/nightly_snapshot.yaml (hunk 4 above)
Comment (NigelBreslaw) [reply .github/workflows/nightly_snapshot.yaml:243]: Good point.

File: package.json (hunk 5 above)
Comment (NigelBreslaw) [reply package.json:2]: whoops

**Comment by tronical** [comment]
I made a nightly build with it:

https://github.com/slint-ui/slint/actions/runs/11213147964/job/31166596221
